plugins/content_distribution/lib/kContentDistributionManager.php

```diff
@@ -1121,4 +1092,47 @@ public static function addAssetIdsToEntryDistribution($entryDistribution, $distr
 		$entryDistribution->save();
 		return $ret_val;
 	}
+
+	private static function checkIfSunriseIsLater($entryDistribution, $distributionProfile)
+	{
+		$sunrise = $entryDistribution->getSunrise(null);
+		if($sunrise)
+		{
+			$distributionProvider = $distributionProfile->getProvider();
+			if(!$distributionProvider->isScheduleUpdateEnabled() && !$distributionProvider->isAvailabilityUpdateEnabled())
+			{
+				$sunrise -= $distributionProvider->getJobIntervalBeforeSunrise();
+				if($sunrise > time())
+					return true;
+			}
+		}
+		return false;
+	}
+
+	private static function updateStatusToQueued($entryDistribution)
+	{
+		if ($entryDistribution->getStatus() != EntryDistributionStatus::IMPORT_SUBMITTING)
+		{
+			$entryDistribution->setStatus(EntryDistributionStatus::QUEUED);
+			$entryDistribution->save();
+		}
+		KalturaLog::info("Will be submitted when ready");
```

Comment:
Add info to the log - Entry %id ,  entryDistribution-> Id%  In status %d will be ........